### DISCUSSION TITLE
Building Locally on Windows

### DISCUSSION THREAD

**Discussion by amura11** [tsightler/ring-mqtt#693] [Support Q&A]
I made some code changes and I wanted to verify them locally but building the Docker image doesn't seem to work correctly. Running the following build command works fine and generates an image:

```
docker build -t ring-mqtt:local-test-1 ./
```

But running that image with the following command results in the error `No such file or directory`:
```
docker run -it --rm --mount type=bind,source=${PWD}\.docker-storage,target=/data --entrypoint /app/ring-mqtt/init-ring-mqtt.js ring-mqtt:local-test-1
```

I thought maybe it was something with my storage but if I replace my image with the prebuilt one it runs fine and doesn't throw an error. I'm assuming the build isn't doing something right but I have no idea.

**Comment by tsightler** [comment]
Do you mean Docker native on Windows?  Unfortunately, I don't know anything about it.  It seems weird seeing the backslash in the path, but perhaps that's OK for Windows.

I reproduced your steps in docker on Linux and there were no issues.  I don't use Docker on Windows, other than via WSL2, so technically still on Linux.  Without logs from the build process, I'm not really clear what the problem would be but note that this project does not support running on Windows, only Linux.

**Reply by amura11** [reply]
I did mean Docker naitively on Windows. It's odd that the container would appear to build (at least no obvious errors) but the image clearly wasn't correct since the same command with the official image would run no problem.

I ended up finding an easier workaround to test the code: running the code locally on Windows. I don't know why I wnet straight to creating my own Docker image but I was able to get the code running so I could test my changes running the following:

```
node .\init-ring-mqtt.js
$env:DEBUG='ring-mqtt';node .\ring-mqtt.js
```

The `init-ring-mqtt.js` did throw an error but it created the state file so ¯\_(ツ)_/¯ good enough to run a few tests and verify my changes before creating a PR.

**Reply by tsightler** [reply]
There are quite a number of functions in ring-mqtt that are not cross-platform friendly, that's why running in node on Windows is not supported.  It might partially work, but there will be lots of breakage in various areas.
